extensions/amp-social-share/1.0/amp-social-share.js

```diff
@@ -158,7 +158,16 @@ class AmpSocialShare extends PreactBaseElement {
 /** @override */
 AmpSocialShare['Component'] = SocialShare;
 
-AmpSocialShare['passthroughNonEmpty'] = true;
+/** @override */
+AmpSocialShare['passthroughNonEmpty'] = dict({
+  'callback': function () {
+    this.scheduleRender_();
```

Comment:
Receiving a closure error here for 'dangerous use of the global this object', am open to suggestions.

Reply:
IMHO this doesn't belong here because two components with `passthroughNonEmpty` should have exactly the same `MutationObserver` setup. I think this should just be a static const map in the `PreactBaseElement` that maps `MutationObserver` options to the embedding type.

Reply:
Sounds good, updated to be in a static map on the `PreactBaseElement`.